### ISSUE TITLE
BD stopped working, attempting to reinstall broke Discord.

### ISSUE THREAD

**Issue by fhuewhd** [bb010g/betterdiscordctl#33]
I've been using BD for a bit, but today when I started Discord it wasn't having any effect - BD was still listed in the optiojns screen, with all settings the way I'd left them, but Discord wasn't following those settings - I'd set it to minimal mode and everything was full size.  I assumed that a Discord update had broken things yet again, and did "betterdiscordctl reinstall -s /usr/share", which is how I usually fix that.
The reply was "Not installed".
I tried "betterdiscordctl install -s /usr/share", and got this:
```
Updating asar...
/home/user/.local/share/betterdiscordctl/asar/node_modules/asar/lib/asar.js:46
    .then(([filenames, metadata]) => module.exports.createPackageFromFiles(src, dest, filenames, metadata, options))
           ^

SyntaxError: Unexpected token [
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:374:25)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Module.require (module.js:354:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/home/user/.local/share/betterdiscordctl/asar/node_modules/asar/bin/asar.js:2:12)
    at Module._compile (module.js:410:26)
    at Object.Module._extensions..js (module.js:417:10)

```

I tried starting Discord again, and after it had shown its loading message for a bit I got this:
```
A fatal Javascript error occurred
Error: Cannot find module './core'
    at Module._resolveFilename (internal/modules/cjs/loader.js:584:15)
    at Function.Module._resolveFilename (/usr/share/discord/resources/electron.asar/common/reset-search-paths.js:43:12)
    at Function.Module._load (internal/modules/cjs/loader.js:510:25)
    at Module.require (internal/modules/cjs/loader.js:640:17)
    at require (internal/modules/cjs/helpers.js:20:18)
    at Object.<anonymous> (/home/user/.config/discord/0.0.8/modules/discord_desktop_core/index.js:1:173)
    at Object.<anonymous> (/home/user/.config/discord/0.0.8/modules/discord_desktop_core/index.js:2:3)
    at Module._compile (internal/modules/cjs/loader.js:693:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:704:10)
    at Module.load (internal/modules/cjs/loader.js:602:32)```
Any idea what's goign on, please?  How can I get my Discord and BD working again?

**Comment by ObserverOfTime** [comment]
Try installing `asar` globally (`sudo npm i -g asar`) and then run
```sh
betterdiscordctl reinstall --global-asar -v -v -s /usr/share
```

**Comment by fhuewhd** [comment]
Thanks for quick reply.
``` betterdiscordctl reinstall --global-asar -v -v -s /usr/share```
gave
```VV: Scanning /usr/share
VV: Trying flavor ''
V: Using Discord at /usr/share/discord
V: Found modules in /home/user/.config/discord/0.0.8/modules
Not installed.```

```betterdiscordctl install --global-asar -v -v -s /usr/share```
gave
```VV: Scanning /usr/share
VV: Trying flavor ''
V: Using Discord at /usr/share/discord
V: Found modules in /home/user/.config/discord/0.0.8/modules
V: Unpacking core asar...
/usr/local/lib/node_modules/asar/lib/asar.js:46
    .then(([filenames, metadata]) => module.exports.createPackageFromFiles(src, dest, filenames, metadata, options))
           ^

SyntaxError: Unexpected token [
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:374:25)
    at Object.Module._extensions..js (module.js:417:10)
    at Module.load (module.js:344:32)
    at Function.Module._load (module.js:301:12)
    at Module.require (module.js:354:17)
    at require (internal/module.js:12:17)
    at Object.<anonymous> (/usr/local/lib/node_modules/asar/bin/asar.js:2:12)
    at Module._compile (module.js:410:26)
    at Object.Module._extensions..js (module.js:417:10)```
and Discord continues not to work.

**Comment by ObserverOfTime** [comment]
This seems like an [asar](/electron/asar) issue. You might wanna submit it there.

**Comment by fhuewhd** [comment]
OK, did that and Discord is now functioning again, and after testing it again more carefully I've worked out what was really happening.  BD itself was never actually broken, or not as broken as I thought (at least, not until I started trying to reinstall it and really did break it).  It wasn't that BD wasn't having any effect at all - it was just that the "minimal mode" option isn't working.  Since I always have minimal mode on, I saw that and assumed that "BD has disappeared".  Feel a bit silly now!  It looks clunky without the minimal mode, but so long as it is doing its job of getting rid of the animations, that'll do for me.
